spec/commands/add_spec.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+require "spec_helper"
+
+describe "bundle add" do
+  before :each do
+    build_repo2
+
+    gemfile <<-G
+      source "file://#{gem_repo2}"
+    G
+  end
+
+  context "when version number is set" do
+    it "adds gem with provided version" do
+      bundle "add activesupport 2.3.5"
+      expect(bundled_app("Gemfile").read).to match(/gem 'activesupport', '= 2.3.5'/)
+    end
+
+    it "adds gem with provided version and version operator" do
+      update_repo2 do
+        build_gem "activesupport", "3.0"
+      end
+
+      bundle "add activesupport '> 2.3.5'"
+      expect(bundled_app("Gemfile").read).to match(/gem 'activesupport', '> 2.3.5'/)
+    end
+  end
+
+  context "when version number is not set" do
+    it "adds gem with last stable version" do
+      bundle "add activesupport"
+      expect(bundled_app("Gemfile").read).to match(/gem 'activesupport', '= 2.3.5'/)
```

Comment:
@indirect I know we discussed making `bundle add` act almost the same as `bundle inject`, but if we're really going to push `bundle add` as a `bundle inject` with far improved developer UX, I think we should have `bundle add` add new gems to the Gemfile with `~>` constraint on the latest version if no version is passed into the command.

I.e.

`bundle add test`

The latest non-beta version of `test` is `v1.0.2`.
The latest pre-release version of `test` is `v1.0.3.rc1`.

`Gemfile`:
```
gem 'test', '~> 1.0.2'
```

`bundle add test --pre`

Gemfile:
```
gem 'test', '~> 1.0.3rc1'
```